lib/source-check.js

```diff
@@ -0,0 +1,25 @@
+'use strict'
+
+const fs = require('fs')
+
+function checkForTranspiledCode (filename) {
+  const readFile = fs.readFileSync(filename, 'utf8')
+  const regex = /function\s+(\w+)/g
+  let matchedObj
+  let isTranspiled = false
+
+  // Check for a source map
+  isTranspiled = readFile.includes('//# sourceMappingURL=')
+
+  while ((matchedObj = regex.exec(readFile)) !== null) {
```

Comment:
We should maybe only do this if the source map comment was not found. There's no possibility that the loop decides that it's actually _not_ transpiled.

Reply:
By default isTranspiled is false.  `let isTranspiled = false`. Either the source check or map comment check will set it to true.

I could change it to :
`If(readFile.includes('//# sourceMappingURL=')) { isTranspiled = true }`

Better?

Reply:
yea, what I mean is that if the source map check has already set it to true, we no longer need to do the loop

Reply:
Yeah, thats right! I'll have another look at it.